integration-tests/src/test/java/tachyon/master/MasterFaultToleranceIntegrationTest.java

```diff
@@ -122,7 +123,42 @@ public void faultTest() throws IOException, TException {
   // TODO: Resolve the issue with HDFS as UnderFS
   @Ignore
   @Test
-  public void createFilesTest() throws IOException, TException {
+  public void deleteFileFaultTest() throws Exception {
+    // Kill leader -> create files -> kill leader -> delete files, repeat.
+    List<Pair<Long, TachyonURI>> answer = Lists.newArrayList();
+    for (int kills = 0; kills < MASTERS - 1; kills ++) {
+      Assert.assertTrue(mLocalTachyonClusterMultiMaster.killLeader());
+      CommonUtils.sleepMs(Constants.SECOND_MS * 2);
+
+      if (kills % 2 != 0) {
+        // Delete files.
+
+        faultTestDataCheck(answer);
+
+        // We can not call mTfs.delete(mTfs.open(new TachyonURI(TachyonURI.SEPARATOR))) because root
+        // node can not be deleted.
+        for (FileInfo file : mTfs.listStatus(mTfs.open(new TachyonURI(TachyonURI.SEPARATOR)))) {
+          mTfs.delete(new TachyonFile(file.getFileId()));
+        }
+        answer.clear();
+        faultTestDataCheck(answer);
+      } else {
+        // Create files.
+
+        Assert.assertEquals(0, answer.size());
+        faultTestDataCheck(answer);
+
+        faultTestDataCreation(new TachyonURI(PathUtils.concatPath(
+            TachyonURI.SEPARATOR, "data_" + kills)), answer);
+        faultTestDataCheck(answer);
+      }
+    }
+  }
+
+  // TODO: Resolve the issue with HDFS as UnderFS
```

Comment:
style: // TODO(cc): Resolve the issue with HDFS as UnderFS.

(tag your TODOs and end sentences with a period)